aten/src/ATen/native/native_functions.yaml

```diff
@@ -170,18 +170,34 @@
     CPU: _assert_async_cpu
     CUDA: _assert_async_cuda
 
+- func: _functional_assert_async(Tensor self, Tensor dep_token) -> Tensor
+  dispatch:
+    CPU: _functional_assert_async_cpu
+
 - func: _assert_async.msg(Tensor self, str assert_msg) -> ()
   dispatch:
     CPU: _assert_async_msg_cpu
     CUDA: _assert_async_msg_cuda
 
+- func: _functional_assert_async.msg(Tensor self, str assert_msg, Tensor dep_token) -> Tensor
+  dispatch:
+    CPU: _functional_assert_async_msg_cpu
+
 - func: _assert_tensor_metadata(Tensor a, SymInt[]? size=None, SymInt[]? stride=None, ScalarType? dtype=None) -> ()
 
 - func: sym_constrain_range(Scalar size, int? min=None, int? max=None) -> ()
   dispatch:
     CPU: sym_constrain_range_cpu
     CUDA: sym_constrain_range_cuda
 
+- func: functional_sym_constrain_range(Scalar size, int? min, int? max, Tensor dep_token) -> Tensor
+  dispatch:
+    CPU: functional_sym_constrain_range_cpu
+
+- func: make_dep_token(Tensor? dep_token) -> Tensor
```

Comment:
Why does it need a optional dep_token?

Reply:
Got some trouble defining op without any argument. But I think got a version which is kind of working. Will update.